src/cogent3/app/data_store.py

```diff
@@ -650,7 +653,10 @@ def __init__(
 
     def _has_other_suffixes(self, path, suffix):
         for f in zipfile.ZipFile(path).namelist():
```

Comment:
refactor this function to be the same as above (define `allowed` as a set, etc..)